Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2734 @@
+        condition = (
+            f'{condition} AND state: {status}'
+            if condition
+            else f'state: {status}'
+        )
+
+    if query:
+        condition = query
+
+    return condition
+
+
+def get_event_filter_query(args: dict[str, Any]) -> str:
+    """
+    This function create the query for search condition as part of response body.
+    Args:
+        args: demisto.args()
+    Returns:
+        Return string.
+    """
+    # Activity query Parameters
+    event_severity_mapping: dict[str, int] = {
+        'info': 1,
+        'warning': 2,
+        'minor': 3,
+        'major': 4,
+        'critical': 5,
+        'fatal': 6
+    }
+
+    event_status_mapping: dict[str, int] = {
+        'Unknown': 0,
+        'Success': 1,
+        'Failure': 2
+    }
+
+    event_type_id = arg_to_number(args.get('type_id'))
+    severity = event_severity_mapping.get(args.get('severity', ''))
+    status = event_status_mapping.get(args.get('status', ''))
+    query = args.get('query')
+
+    if query and (event_type_id or severity):
+        raise DemistoException(INVALID_QUERY_ERROR_MSG)
+
+    condition = ''
+    if event_type_id:
+        condition = f'type_id: {event_type_id}'
+
+    if severity:
+        condition = (
+            f'{condition} AND severity_id: {severity}'
+            if condition
+            else f'severity_id: {severity}'
+        )
+
+    if status:
+        condition = (
+            f'{condition} AND status_id: {status}'
+            if condition
+            else f'status_id: {status}'
+        )
+
+    if query:
+        condition = query
+
+    return condition
+
+
+def get_association_filter_query(args: dict) -> str:
+    """
+    This function validate the association filter search query and create the query search condition a
+    payload based on the demisto.args().
+    Args:
+        args: demisto.args()
+    Returns:
+        Return string.
+    """
+    query_type = args.get('search_object', '')
+    query_value = args.get('search_value', '')
+    query = args.get('query', '')
+
+    if query_type and query_type not in SEARCH_QUERY_TYPE:
+        raise DemistoException(f'Invalid Search Type! Only supported type are : {SEARCH_QUERY_TYPE}')
+
+    if query and (query_type or query_value):
+        raise DemistoException(INVALID_QUERY_ERROR_MSG)
+
+    if query_type == 'sha256':
+        condition = query_search_condition('sha256', query_value)
+        return f'sha2: ({condition})'
+    elif query_type == 'device_uid':
+        condition = query_search_condition('device_uid', query_value, ignore_validation=True)
+        return f'device_uid: ({condition})'
+    elif query_type == 'domain':
+        condition = query_search_condition('domain', query_value, ignore_validation=True)
+        return f'data_source_url_domain: ({condition})'
+    else:
+        return query
+
+
+# def post_request_body(args: dict, limit: int = 1) -> dict:
+def post_request_body(args: dict) -> dict[str, Any]:
```

Comment:
remove this one please. All requests should be via client methods.

Reply:
I have removed the line no 1413. I think function name is incorrect, therefore rename the function "create_payload_from_argument" only. I have limited the changes further as all the endpoints client methods required to pass payload argument.